### ISSUE TITLE
MPI thread - AMReX

### ISSUE THREAD

**Issue by LDAmorim** [ECP-WarpX/WarpX#1380]
Hi all,

I asked @ax3l for help with this yesterday and he already fix it in https://github.com/AMReX-Codes/amrex/pull/1406.

But I wanted to open (and close right after) an issue here for future notice.
Just so you would remember to update your `AMReX` repo and for you to know what kind of error message to expect.
(I had a hard time finding it's originating code inside `AMReX` repo/docs.)

I found the following error message when running `WarpX` on CORI@NERSC with the submitjob based on `WarpX`'s docs for CORI and `openPMD`:
```
MPI provided < requested: MPI_THREAD_SERIALIZED < MPI_THREAD_MULTIPLE
```
Which was due to `AMReX` having checked for multiple threading without initializing it (if I understood the fixed lines correctly).

Thanks again for your help with this @ax3l and thank you @WeiqunZhang for merging it already!

Cheers,
Diana

----

```
#SBATCH -N 1
#SBATCH -t 00:10:00
#SBATCH -q debug
#SBATCH -C knl
#SBATCH -J G4RZLongBeam
#SBATCH -A m2852
#SBATCH -e error.txt
#SBATCH -o output.txt
#SBATCH -S 4
module load craype-mic-knl
module load PrgEnv-gnu
module load cmake/3.14.4
module load cray-hdf5-parallel
module load adios2/2.5.0
export PKG_CONFIG_PATH=$PWD/../openPMD-install/lib64/pkgconfig:$PKG_CONFIG_PATH
export CMAKE_PREFIX_PATH=$PWD/../openPMD-install:$CMAKE_PREFIX_PATH
export OMP_PLACES=threads
export OMP_PROC_BIND=spread
# KNLs have 4 hyperthreads max
export CORI_MAX_HYPETHREAD_LEVEL=4
# We use 64 cores out of the 68 available on Cori KNL,
# and leave 4 to the system (see "#SBATCH -S 4" above).
export CORI_NCORES_PER_NODE=64
# Typically use 8 MPI ranks per node without hyperthreading,
# i.e., OMP_NUM_THREADS=8
export WARPX_NMPI_PER_NODE=8
export WARPX_HYPERTHREAD_LEVEL=1
# Compute OMP_NUM_THREADS and the thread count (-c option)
export CORI_NHYPERTHREADS_MAX=$(( ${CORI_MAX_HYPETHREAD_LEVEL} * ${CORI_NCORES_PER_NODE} ))
export WARPX_NTHREADS_PER_NODE=$(( ${WARPX_HYPERTHREAD_LEVEL} * ${CORI_NCORES_PER_NODE} ))
export OMP_NUM_THREADS=$(( ${WARPX_NTHREADS_PER_NODE} / ${WARPX_NMPI_PER_NODE} ))
export WARPX_THREAD_COUNT=$(( ${CORI_NHYPERTHREADS_MAX} / ${WARPX_NMPI_PER_NODE} ))
date
srun --cpu_bind=cores -n $(( ${SLURM_JOB_NUM_NODES} * ${WARPX_NMPI_PER_NODE} )) -c ${WARPX_THREAD_COUNT} /global/homes/l/ligiada/WarpX/WarpX/Bin/main2d.gnu.mic-knl.TPROF.MTMPI.OMP.OPMD.RZ.ex input
date
exit
```

**Comment by LDAmorim** [comment]
Closed in AMReX-Codes/amrex#1406

**Comment by ax3l** [comment]
Thanks for documenting this temporary issue. Yes, you got that PR in AMReX right.
Also related to #1376
